astropy/table/table.py

```diff
@@ -2521,7 +2521,22 @@ def read(cls, *args, **kwargs):
         The arguments and keywords (other than ``format``) provided to this function are
         passed through to the underlying data reader (e.g. `~astropy.io.ascii.read`).
         """
-        return io_registry.read(cls, *args, **kwargs)
+        # Always read as Table because all astropy readers initially return
+        # Table or QTable.  Most readers return Table but e.g. ascii.ecsv might
+        # return QTable.
+        out = io_registry.read(Table, *args, **kwargs)
```

Comment:
I understand the post-read conversion, but not quite why the input to registry read should be changed -- won't many readers do it correctly already?  Indeed, if I undo this locally, tests still pass.